pinot-core/src/test/java/org/apache/pinot/queries/ArgMinMaxTest.java

```diff
@@ -0,0 +1,583 @@
+        "argmin(case(equals(stringColumn,'a33'),equals(stringColumn,'a22'),'b','a','c'),stringColumn)");
+  }
+
+  @Test
+  public void testGroupByInterSegment() {
+    // Simple inter segment group by
+    String query = "SELECT groupByIntColumn, arg_max(intColumn, longColumn) FROM testTable GROUP BY groupByIntColumn";
+
+    BrokerResponseNative brokerResponse = getBrokerResponse(query);
+    ResultTable resultTable = brokerResponse.getResultTable();
+    List<Object[]> rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 10);
+
+    assertEquals(rows.get(0)[0], 1);
+    assertEquals(rows.get(0)[1], 996L);
+
+    assertNull(rows.get(1)[0]);
+    assertEquals(rows.get(1)[1], 996L);
+
+    assertNull(rows.get(9)[0]);
+    assertEquals(rows.get(9)[1], 995L);
+
+    assertEquals(rows.get(6)[0], 4);
+    assertEquals(rows.get(6)[1], 999L);
+
+    // Simple inter segment group by with limit
+    query =
+        "SELECT groupByIntColumn2, arg_max(longColumn, doubleColumn) FROM testTable GROUP BY groupByIntColumn2 ORDER "
+            + "BY groupByIntColumn2 LIMIT 15";
+
+    brokerResponse = getBrokerResponse(query);
+    resultTable = brokerResponse.getResultTable();
+    rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 24);
+
+    assertEquals(rows.get(0)[0], 1);
+    assertEquals(rows.get(0)[1], 0D);
+
+    assertNull(rows.get(1)[0]);
+    assertEquals(rows.get(1)[1], 0D);
+
+    assertEquals(rows.get(22)[0], 2048);
+    assertEquals(rows.get(22)[1], 1999D);
+
+    assertNull(rows.get(23)[0]);
+    assertEquals(rows.get(23)[1], 1999D);
+
+    // MV inter segment group by
+    query = "SELECT groupByMVIntColumn, arg_min(intColumn, doubleColumn) FROM testTable GROUP BY groupByMVIntColumn";
+
+    brokerResponse = getBrokerResponse(query);
+    resultTable = brokerResponse.getResultTable();
+    rows = resultTable.getRows();
+
+    assertEquals(rows.size(), 20);
+
+    assertEquals(rows.get(0)[0], 1);
+    assertEquals(rows.get(0)[1], 0D);
+
+    assertNull(rows.get(1)[0]);
+    assertEquals(rows.get(1)[1], 0D);
+
+    assertEquals(rows.get(2)[0], 2);
+    assertEquals(rows.get(2)[1], 1D);
+
+    assertNull(rows.get(3)[0]);
+    assertEquals(rows.get(3)[1], 1D);
+
+    assertEquals(rows.get(16)[0], 9);
+    assertEquals(rows.get(16)[1], 8D);
+
+    assertNull(rows.get(17)[0]);
+    assertEquals(rows.get(17)[1], 8D);
+
+    assertEquals(rows.get(18)[0], 0);
+    assertEquals(rows.get(18)[1], 0D);
+
+    assertNull(rows.get(19)[0]);
+    assertEquals(rows.get(19)[1], 0D);
+
+    // MV inter segment group by with projection on MV column
+    query = "SELECT groupByMVIntColumn, arg_min(intColumn, mvIntColumn), "
+        + "arg_max(intColumn, mvStringColumn) FROM testTable GROUP BY groupByMVIntColumn";
+
+    brokerResponse = getBrokerResponse(query);
+    resultTable = brokerResponse.getResultTable();
+    rows = resultTable.getRows();
+    assertEquals(rows.size(), 20);
+
+    assertEquals(rows.get(18)[0], 0);
```

Comment:
nit: here and other tests in this file, is there any value in checking the values returned for a few more data points (here you only check row 18)? or possibly creating a static array of expected result and looping over it to check the actual against expected. not necessary to add right now or to add if it doesn't make sense. just wondering if it does.

Reply:
enhanced most of them, thanks